darts/utils/model_selection.py

```diff
@@ -0,0 +1,235 @@
+            if self.type == 'train':
+                return split_index
+            else:
+                return len(self.data) - split_index
+        else:
+            return len(self.data)
+
+    @classmethod
+    def make_splitter(cls,
+                      data: Union[TimeSeries, Sequence[TimeSeries]],
+                      test_size: Optional[Union[float, int]] = 0.25,
+                      axis: Optional[int] = 0,
+                      input_size: Optional[int] = 0,
+                      horizon: Optional[int] = 0,
+                      vertical_split_type: Optional[str] = SIMPLE,
+                      lazy: bool = False
+                      ) -> Union[Tuple[TimeSeries, TimeSeries], Tuple[Sequence[TimeSeries], Sequence[TimeSeries]]]:
+
+        if not isinstance(data, Sequence):
+            axis = 1
+            data = [data]  # convert to sequence for unified processing later
+            single_timeseries = True
+        else:
+            single_timeseries = False
+
+        train_set = cls(type='train', data=data, test_size=test_size, axis=axis, input_size=input_size, horizon=horizon,
+                        vertical_split_type=vertical_split_type)
+
+        test_set = cls(type='test', data=data, test_size=test_size, axis=axis, input_size=input_size, horizon=horizon,
+                       vertical_split_type=vertical_split_type)
+
+        if single_timeseries:
+            return train_set[0], test_set[0]
+        else:
+            if lazy:
+                return train_set, test_set
+            else:
+                return list(train_set), list(test_set)
+
+
+def train_test_split(
+        data: Union[TimeSeries, Sequence[TimeSeries]],
+        test_size: Optional[Union[float, int]] = 0.25,
+        axis: Optional[int] = 0,
+        input_size: Optional[int] = 0,
+        horizon: Optional[int] = 0,
+        vertical_split_type: Optional[str] = SIMPLE,
+        lazy: bool = False
+        ) -> Union[Tuple[TimeSeries, TimeSeries], Tuple[Sequence[TimeSeries], Sequence[TimeSeries]]]:
+    """
+    Splits the dataset into training and test dataset. Supports splitting along the sample axis and time axis.
+
+    If the input type is single TimeSeries, then only splitting over time axis is available, thus ``input_size`` and
+    ``horizon`` have to be provided.
+
+    When splitting over the time axis, splitter tries to greedy satisfy the requested test set size, i.e. when one of
+    the timeseries in sequence is too small, all samples will go to the test set and the exception will be raised.
+
+
+    Time axis split with ``model-aware`` split type enabled tries to reclaim as much datapoints for training as
+    possible by partially overlapping test set with training set. This is possible because only forecasted part of
+    test set cannot be used for training. The formula to calculate last available timestep for training set is
+    following:
+
+        train end index = timeseries length - horizon
```

Comment:
```suggestion
        train end index = timeseries length - horizon - test_size
```
According to line 91, train_end_index = ts_length - self.horizon - test_size + 1. Since the comment mention the last _available_ index, then the +1 should not be necessary in this formula